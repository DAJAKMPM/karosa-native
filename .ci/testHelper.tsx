import React, { JSXElementConstructor } from "react";
import { View } from "react-native";
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from "@testing-library/react-native";
import { Formik, FormikValues } from "formik";

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  _themeProp: unknown = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };

  const renderApi = render(children, options);
  const wrapper = renderApi.queryByTestId(wrapperTestID || "wrapper");
  return { wrapper, ...renderApi };
};

export const renderWithFormikWrapper = <T extends FormikValues>(
  initialValues: T,
  children: React.ReactElement<any, string | JSXElementConstructor<any>>
) => {
  return renderWithWrapper(
    <Formik
      initialValues={initialValues}
      onSubmit={() => console.log("submit")}
      component={() => children}
    />
  );
};
