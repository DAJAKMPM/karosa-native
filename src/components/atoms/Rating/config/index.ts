/**
 *
 * @format
 *
 */

export const defaultRatingCount = 5;
