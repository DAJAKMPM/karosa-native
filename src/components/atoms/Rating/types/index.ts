/**
 *
 * @format
 *
 */

export type PropsType = {
  ratingCount?: number;
  imageSize?: number;
  startingValue?: number;
  fractions?: number;
  minValue?: number;
  readonly?: boolean;
  showRating?: boolean;
  onFinishRating?(rating: number): void;
  onStartRating?(): void;
};
