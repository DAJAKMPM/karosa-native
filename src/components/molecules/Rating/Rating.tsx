/**
 *
 * Rating
 * @format
 *
 */

import React, { FC, useState } from "react";
import { View, Image, TouchableOpacity } from "react-native";

import type { PropsType } from "./types";
import RatingStyles from "./styles";

const Rating: FC<PropsType> = (props) => {
  const { rate } = props;
  const [defaultRating, setDefaultRating] = useState(rate);
  const [maxRating, setMaxRating] = useState([1, 2, 3, 4, 5]);
  const starImageFilled =
    "https://raw.githubusercontent.com/AboutReact/sampleresource/master/star_filled.png";
  const starImageCorner =
    "https://raw.githubusercontent.com/AboutReact/sampleresource/master/star_corner.png";

  return (
    <View style={RatingStyles.customRatingBarStyle}>
      {maxRating.map((item) => {
        return (
          <View>
            <Image
              style={RatingStyles.starImageStyle}
              source={
                item <= defaultRating
                  ? { uri: starImageFilled }
                  : { uri: starImageCorner }
              }
            />
          </View>
        );
      })}
    </View>
  );
};

export default Rating;
