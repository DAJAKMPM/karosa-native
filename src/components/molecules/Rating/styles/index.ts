/**
 *
 * @format
 *
 */

import { StyleSheet } from "react-native";

const RatingStyles = StyleSheet.create({
  customRatingBarStyle: {
    margin: 3,
    justifyContent: "center",
    flexDirection: "row",
  },
  starImageStyle: {
    width: 16,
    height: 16,
  },
  container: {
    height: 30,
  },
});

export default RatingStyles;
