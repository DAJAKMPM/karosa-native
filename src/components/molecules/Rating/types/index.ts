/**
 *
 * @format
 *
 */

export type PropsType = {
  rate: number;
};
