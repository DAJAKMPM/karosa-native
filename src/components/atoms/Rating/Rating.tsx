/**
 *
 * Rating
 * @format
 *
 */

import React, { FC } from "react";
import { Rating as RnRating } from "react-native-elements";

import { defaultRatingCount } from "./config";
import type { PropsType } from "./types";

const Rating: FC<PropsType> = (props) => {
  const {
    ratingCount = defaultRatingCount,
    fractions,
    imageSize,
    minValue,
    onFinishRating,
    onStartRating,
    readonly,
    showRating,
    startingValue,
  } = props;

  return (
    <RnRating
      type="star"
      ratingCount={ratingCount}
      readonly={readonly}
      imageSize={imageSize}
      startingValue={startingValue}
      onFinishRating={onFinishRating}
      onStartRating={onStartRating}
      fractions={fractions}
      showRating={showRating}
      minValue={minValue}
    />
  );
};

export default Rating;
