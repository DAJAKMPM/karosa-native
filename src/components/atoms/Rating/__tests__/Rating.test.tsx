import React from "react";
import { renderWithWrapper } from "@app/ci/testHelper";
import Rating from "../Rating";

const FAKE_URI = "https://i.imgur.com/0y8Ftya.jpg";

describe("Radio Component", () => {
  it("should match snapshot", () => {
    const { toJSON } = renderWithWrapper(
      <Rating
        type="custom"
        ratingCount={5}
        readonly
        imageSize={10}
        startingValue={5}
        ratingImage={{
          source: {
            uri: FAKE_URI,
          },
        }}
      />
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
