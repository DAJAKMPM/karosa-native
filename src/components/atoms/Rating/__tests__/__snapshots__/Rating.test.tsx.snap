// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Component should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <View
    pointerEvents="none"
  >
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#f1c40f",
              "height": 10,
              "width": 50,
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "white",
              "height": 10,
              "width": 0,
            }
          }
        />
      </View>
      <View>
        <Image
          source={
            Object {
              "source": Object {
                "uri": "https://i.imgur.com/0y8Ftya.jpg",
              },
            }
          }
          style={
            Object {
              "height": 10,
              "tintColor": undefined,
              "width": 10,
            }
          }
        />
      </View>
      <View>
        <Image
          source={
            Object {
              "source": Object {
                "uri": "https://i.imgur.com/0y8Ftya.jpg",
              },
            }
          }
          style={
            Object {
              "height": 10,
              "tintColor": undefined,
              "width": 10,
            }
          }
        />
      </View>
      <View>
        <Image
          source={
            Object {
              "source": Object {
                "uri": "https://i.imgur.com/0y8Ftya.jpg",
              },
            }
          }
          style={
            Object {
              "height": 10,
              "tintColor": undefined,
              "width": 10,
            }
          }
        />
      </View>
      <View>
        <Image
          source={
            Object {
              "source": Object {
                "uri": "https://i.imgur.com/0y8Ftya.jpg",
              },
            }
          }
          style={
            Object {
              "height": 10,
              "tintColor": undefined,
              "width": 10,
            }
          }
        />
      </View>
      <View>
        <Image
          source={
            Object {
              "source": Object {
                "uri": "https://i.imgur.com/0y8Ftya.jpg",
              },
            }
          }
          style={
            Object {
              "height": 10,
              "tintColor": undefined,
              "width": 10,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
